### PR TITLE
CxIAST integration documentation

### PR DESCRIPTION
**Pull request by yevgenykuz**
### Description

CxIAST integration documentation. See #725 for actual code.

### References

#725 - actual code.

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentation is a Wiki Update, please indicate desired changes within PR MD Comment*
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 1412 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**44 Medium**<br />Low&nbsp;**1262 Low**<br />Info&nbsp;**106 Info**<br />
### Violation Summary
Medium&nbsp;**10** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1059533&projectid=2118)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[67](https://github.com/checkmarx-ltd/cx-flow/blob/iast-docs/src/main/java/com/checkmarx/flow/controller/FlowController.java#L67) [70](https://github.com/checkmarx-ltd/cx-flow/blob/iast-docs/src/main/java/com/checkmarx/flow/controller/FlowController.java#L70) [71](https://github.com/checkmarx-ltd/cx-flow/blob/iast-docs/src/main/java/com/checkmarx/flow/controller/FlowController.java#L71) [76](https://github.com/checkmarx-ltd/cx-flow/blob/iast-docs/src/main/java/com/checkmarx/flow/controller/FlowController.java#L76) [78](https://github.com/checkmarx-ltd/cx-flow/blob/iast-docs/src/main/java/com/checkmarx/flow/controller/FlowController.java#L78) [137](https://github.com/checkmarx-ltd/cx-flow/blob/iast-docs/src/main/java/com/checkmarx/flow/controller/FlowController.java#L137) |Medium|Input_Path_Not_Canonicalized|src/main/java/com/checkmarx/flow/controller/FlowController.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1059533&projectid=2118&pathid=23)
[213](https://github.com/checkmarx-ltd/cx-flow/blob/iast-docs/src/main/java/com/checkmarx/flow/utils/ScanUtils.java#L213) |Medium|Input_Path_Not_Canonicalized|src/main/java/com/checkmarx/flow/utils/ScanUtils.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1059533&projectid=2118&pathid=7)
[100](https://github.com/checkmarx-ltd/cx-flow/blob/iast-docs/src/main/java/com/checkmarx/flow/controller/GitHubController.java#L100) [242](https://github.com/checkmarx-ltd/cx-flow/blob/iast-docs/src/main/java/com/checkmarx/flow/controller/GitHubController.java#L242) |Medium|Input_Path_Not_Canonicalized|src/main/java/com/checkmarx/flow/controller/GitHubController.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1059533&projectid=2118&pathid=11)
[60](https://github.com/checkmarx-ltd/cx-flow/blob/iast-docs/src/main/java/com/checkmarx/flow/service/ExternalScriptService.java#L60) |Medium|Input_Path_Not_Canonicalized|src/main/java/com/checkmarx/flow/service/ExternalScriptService.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1059533&projectid=2118&pathid=50)

File: docs/CxIAST-Integration.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+* [Suggested Flow](#suggestedFlow)
```
Comment (jbrotsos) [inline docs/CxIAST-Integration.md:1]: Rename Suggested Flow to "Overview"

Comment (jbrotsos) [inline docs/CxIAST-Integration.md:1]: Filename should be "CxIAST Integration"

File: docs/CxIAST-Integration.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+* [Configuration](#configuration)
+* [Filters](#filters)
+* [Thresholds](#thresholds)
+* [Bug Trackers](#bugTrackers)
+
+## <a name="suggestedFlow">Suggested Flow</a>
+CxIAST can be integrated within a CI/CD pipeline using CxFlow.  
+A CxIAST scan starts automatically when an application with an attached CxIAST agent is being tested.  
+CxIAST scans can be stopped in two ways:
```
Comment (jbrotsos) [inline docs/CxIAST-Integration.md:11]: CxIAST can be integrated within a CI/CD pipeline using CxFlow.  CxIAST scans start automatically when an application with an attached CxIAST agent is being tested.  To attach a CxIAST agent, please follow the following steps: [Insert steps here].
  
CxIAST scans can be stopped in two ways:

File: docs/CxIAST-Integration.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+1. Manually stopped via the CxIAST manager.
+2. By terminating the running application.
+
+Onces tests are done, CxFlow is used to stop the running CxIAST scan.  
+Then, CxFlow will collect the results from CxIAST, analyze them, and open tasks/tickets in the configured bug tracker.  
+To make sure the correct scan is stopped by CxFlow, CxIAST scan tags are used:
```
Comment (jbrotsos) [inline docs/CxIAST-Integration.md:17]: Onces tests are completed, CxFlow is used to stop the running CxIAST scan.  CxFlow will then collect the results from CxIAST, analyze them, and open tasks/tickets in the configured Bug Tracker [put link to Bug Tracker here].  To make sure the correct scan is stopped by CxFlow, CxIAST scan tags are used

File: docs/CxIAST-Integration.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  url: http://xxxxx.iast.net
+  manager-port: 8380
+  username: xxxx
+  password: xxxx
+  update-token-seconds: 150  # CxAccessControl token timeout
+```
+
+## <a name="filters">Filters - to be implemented</a>
+CxFlow can filter CxIAST vulnerabilities before creating bug tracker tickets.  
+The filtering is done by vulnerability severity.
```
Comment (jbrotsos) [inline docs/CxIAST-Integration.md]: CxFlow filters CxIAST vulnerabilities by severity before creating bug tracker tickets.

**Review by jbrotsos — CHANGES_REQUESTED** [review]

**Review by jbrotsos — APPROVED** [review]

**Comment by jbrotsos** [comment]
@yevgenykuz to get the tests to run, the branch name has to begin with pr-

**Comment by yevgenykuz** [comment]
Closing in favor of #754.

**Pull request closed without merging**